tests/kernel/mem_protect/mem_map/testcase.yaml

```diff
@@ -2,3 +2,17 @@ tests:
   kernel.memory_protection.mem_map:
     tags: kernel mmu ignore_faults
     filter: CONFIG_MMU
+    extra_sections: _TRANSPLANTED_FUNC
+    platform_exclude: qemu_x86_64
+  kernel.memory_protection.mem_map.qemu_x86_64:
```

Comment:
not sure why we need 2 scenarios for x64?

Reply:
Because changing the memory model also changes how code is generated. This is an attempt to make sure that tests other than the execution test still test code generated when building without the compiler switch.